### PR TITLE
🌐 [i18n-KO] Translated `testing.md` to Korean

### PR DESCRIPTION
**Pull request by Sunmin0520**
# What does this PR do?

Translated the `testing.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [x] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)
@kihoon71, @0525hhgus, @54data, @seank021, @augustinLib

## Who can review? (Final)
May you please review this PR? @sgugger, @ArthurZucker, @eunseojo

### CODE REVIEW

**Review by 0525hhgus — CHANGES_REQUESTED** [review]
긴 글이었지만, 주석까지 꼼꼼하게 번역해주셨군요! 
좋은 번역 감사합니다 👍
TOC와 용어 관련하여 리뷰를 남겼으니, 확인 부탁드립니다 :)

**Comment by seank021** [comment]
위에 현서 멘토님께서 올려주신 것 외에는 리뷰 사항 없습니다!

File: docs/source/ko/testing.md [hunk 1]
```diff
@@ -0,0 +1,1278 @@
...
+
+
+먼저 🤗 Transformers 모델이 어떻게 테스트되는지 살펴보고, 새로운 테스트를 작성 및 기존 테스트를 개선하는 방법을 알아봅시다.
+
+이 저장소에는 2개의 테스트 스위트가 있습니다:
+
+1. `tests` - 일반 API에 대한 테스트
+2. `examples` - API의 일부가 아닌 다양한 응용 프로그램에 대한 테스트
+
+## Transformers 테스트 방법[[how transformers are tested]]
```
[the remark this answers is not in the document]
Comment (Sunmin0520) [reply docs/source/ko/testing.md:27]: 확인 감사합니다. 알려주신 것처럼 수정하겠습니다:)

File: docs/source/ko/testing.md [hunk 2]
```diff
@@ -0,0 +1,1278 @@
...
+
+🤗 transformers 테스트는 대부분 `unittest`를 기반으로 하지만, 
+`pytest`에서 실행되므로 대부분의 경우 두 시스템의 기능을 사용할 수 있습니다.
+
+지원되는 기능에 대해 [여기](https://docs.pytest.org/en/stable/unittest.html)에서 확인할 수 있지만, 
+기억해야 할 중요한 점은 대부분의 `pytest` fixture가 작동하지 않는다는 것입니다.
+파라미터화도 작동하지 않지만, 우리는 비슷한 방식으로 작동하는 `parameterized` 모듈을 사용합니다.
+
+
+### 파라미터화[[parametrization]]
```
[the remark this answers is not in the document]
Comment (Sunmin0520) [reply docs/source/ko/testing.md:585]: 제가 보기에도 매개변수화가 조금 더 자연스러워서 그렇게 수정하겠습니다!

File: docs/source/ko/testing.md [hunk 3]
```diff
@@ -0,0 +1,1278 @@
...
+이러한 테스트를 실행하려면 적어도 2개의 GPU가 필요합니다.
+
+```bash
+CUDA_VISIBLE_DEVICES=0,1 RUN_SLOW=1 pytest -sv tests/test_trainer_distributed.py
+```
+
+### 출력 캡처[[output-capture]]
+
+테스트 실행 중 `stdout` 및 `stderr`로 전송된 모든 출력이 캡처됩니다. 
+테스트나 설정 메서드가 실패하면 캡처된 출력은 일반적으로 실패 추적 정보와 함께 표시됩니다.
```
Comment (54data) [inline docs/source/ko/testing.md:536]:
```suggestion
테스트나 설정 메소드가 실패하면 캡처된 출력은 일반적으로 실패 추적 정보와 함께 표시됩니다.
```
Glossary에 따라 메서드 -> 메소드로 제안드립니다!

File: docs/source/ko/testing.md [hunk 4]
```diff
@@ -0,0 +1,1278 @@
...
+        env = self.get_env()
+        # 이제 `env`를 사용하여 외부 프로그램 호출
+```
+
+테스트 파일이 `tests` 테스트 스위트 또는 `examples`에 있는지에 따라 
+`env[PYTHONPATH]`가 두 디렉터리 중 하나를 포함하도록 설정되며, 
+현재 저장소에 대해 테스트가 수행되도록 `src` 디렉터리도 포함됩니다.
+테스트 호출 이전에 설정된 경우에는 `env[PYTHONPATH]`를 그대로 사용합니다.
+
+이 헬퍼 메서드는 `os.environ` 객체의 사본을 생성하므로 원본은 그대로 유지됩니다.
```
Comment (54data) [inline docs/source/ko/testing.md:1164]:
```suggestion
이 헬퍼 메소드는 `os.environ` 객체의 사본을 생성하므로 원본은 그대로 유지됩니다.
```

File: docs/source/ko/testing.md (hunk 3 above)
Comment (Sunmin0520) [reply docs/source/ko/testing.md]: 감사합니다. Glossary에 따라 수정했습니다 :)

**Review by sgugger — APPROVED** [review]
Thanks!

**Pull request merged**